### PR TITLE
Better empty spec error message

### PR DESCRIPTION
**Pull request by videlov**
**Description**

Changes proposed in this pull request:

- Better empty spec error message

**Pre-Merge Checklist**

- [ ] As a PR reviewer, verify code coverage and evaluate if it is acceptable.

**Related issues**

### CODE REVIEW

**Review by nataliasitko — APPROVED** [review]

**Review by barchw — CHANGES_REQUESTED** [review]

File: internal/validation/v2alpha1/v2alpha1.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func (a *APIRuleValidator) Validate(ctx context.Context, client client.Client, v
 	if reflect.DeepEqual(a.ApiRule.Spec, gatewayv2alpha1.APIRuleSpec{}) {
 		failures = append(failures, validation.Failure{
 			AttributePath: ".spec",
-			Message:       "APIRule in v2alpha1 version has an empty spec, check the troubleshooting guide",
+			Message:       "APIRule in version v2alpha1 contains an empty spec. To troubleshoot, see https://kyma-project.io/#/api-gateway/user/troubleshooting-guides/03-70-empty-apirule-spec.",
```
[the remark this answers is not in the document]
Comment (videlov) [reply internal/validation/v2alpha1/v2alpha1.go:29]: yes, this is quite in the grayish area of nice, I would also avoid it. we just try to be a bit more helpful if someone is actually checking error state messages. configurable is a bit too much, will move it to a constant.

[the remark this answers is not in the document]
Comment (videlov) [reply internal/validation/v2alpha1/v2alpha1.go]: @nataliasitko any input on this? how can we ensure link is still valid and not outdated, what is best practice?

[the remark this answers is not in the document]
Comment (nataliasitko) [reply internal/validation/v2alpha1/v2alpha1.go]: I agree with you. I didn't think about the fact that when a link breaks, we won't spot it. I asked other TWs how they handle links in error messages, as I'm not aware of any tool to check this

[the remark this answers is not in the document]
Comment (nataliasitko) [reply internal/validation/v2alpha1/v2alpha1.go]: I learned that [Huskies also add links to error messages](https://github.com/kyma-project/telemetry-manager/blob/424444fb73c5b22507e47e3b1db53d60b53d18b8/internal/conditions/conditions.go#L74). Some teams have custom link checks in their repos (besides the general [md-link-check](https://github.com/kyma-project/md-check-link/actions)) but I'm not sure if they also check non-md files
Alternatively, maybe we could link to the troubleshooting folder? It's very unlikely this link will get broken: https://kyma-project.io/#/api-gateway/user/troubleshooting-guides/README

[the remark this answers is not in the document]
Comment (barchw) [reply internal/validation/v2alpha1/v2alpha1.go]: I guess we can add that, and we could have some backlog item to add a linter that would check for deadlinks
Running a simple regex search for `kyma-project.io` links and then checking if something is behind them should be fairly easy.

**Review by barchw — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Review by barchw — APPROVED** [review]

**Review by barchw — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**